broadinstitute/single_cell_portal_core
# ensuring client-server sync on cluster colors (SCP-2943)

SCP-2943.  Clicking around, there were sometimes cases where the client and server would get out-of-sync as to the custom colors.  this change ensures the cluster_file_info and custom_colors are always included in the server response, and also adds better nil-safing for edge cases.

TO TEST:
 1. go to a study you have edit permissions
 2. click around a whole bunch customizing colors, cancelling, reset to defaults, set again etc...
 3. check that the plot, legend, and server color state are in sync
